### ISSUE TITLE
Want Redirect to take effect on a specific domain but not in certain special cases

### ISSUE THREAD

**Issue by KamiyaMinoru** [einaregilsson/Redirector#375]
Example: https://www.dcard.tw/f/3c
pattern: https://www.dcard.tw/f/([a-z0-9].+)
Redirect to: https://www.dcard.tw/f/$1?tab=latest
can lead to https://www.dcard.tw/f/3c?tab=latest, in the case

but it also affects https://www.dcard.tw/f/3c/p/253974524
p.s. /p/[0-9]+ = post number when I click into certain post
which means it will still get https://www.dcard.tw/f/3c?tab=latest
![image](https://github.com/einaregilsson/Redirector/assets/78710607/1e813fcd-e449-449e-8ea7-8b64d8639df5)

while my target is "when the address includes /p/ it does not redirect anything

I solved it via pattern: https://www.dcard.tw/f/([a-zA-Z0-9]*$)+

![image](https://github.com/einaregilsson/Redirector/assets/78710607/991b0a04-7ee4-4b1c-b3e1-5f64d0fd8199)
![image](https://github.com/einaregilsson/Redirector/assets/78710607/42ae81b8-8158-4843-8297-e59623c3560a)

but actually I have no idea what's the problem with the 1st set (why it does not stop on "/" ?)

**Comment by KamiyaMinoru** [comment]
Also, I refer to this example:
![image](https://github.com/einaregilsson/Redirector/assets/78710607/24f15ee2-6fdd-4aca-9636-7c1514bf64be)

but it does not exclude the 2nd folder, seems...
![image](https://github.com/einaregilsson/Redirector/assets/78710607/4118580d-5f1c-4515-ae33-36f63e252127)

**Comment by jadedgnome** [comment]
the regular expression ".+" in the first pattern matches any character including "/" until the end of the string. 

"." represents any character and the "+" that follows it means match 1 or more occurrence. 
tldr: you told it to match everything after "https://www.dcard.tw/f/([a-z0-9]"

**Comment by KamiyaMinoru** [comment]
> the regular expression ".+" in the first pattern matches any character including "/" until the end of the string.
> 
> "." represents any character and the "+" that follows it means match 1 or more occurrence of the previous character. tldr: you told it to match everything after "[https://www.dcard.tw/f/([a-z0-9]](https://www.dcard.tw/f/(%5Ba-z0-9%5D)"

Hi, I used "+" (repeat) but not "."

**Comment by KamiyaMinoru** [comment]
finally I wrote:
redirect: `https://www.dcard.tw/f/([a-zA-Z0-9]*$)+`
to: `https://www.dcard.tw/f/$1?tab=latest`
and solved this issue. Thanks.
